x-pack/plugins/lens/public/visualizations/xy/xy_config_panel/annotations_config_panel/annotations_panel.tsx

```diff
@@ -30,10 +29,9 @@ export const AnnotationsPanel = (
 ) => {
   const { state, setState, layerId, accessor, frame } = props;
 
-  const { inputValue: localState, handleInputChange: setLocalState } = useDebouncedValue<XYState>({
-    value: state,
-    onChange: setState,
-  });
+  // we don't listen to the state prop after the initial render, because we don't want to
+  // slow the annotation settings UI updates down on a full Redux state update
+  const [localState, setLocalState] = useState<XYState>(state);
```

Comment:
This is the key. Instead of relying on `useDebouncedValue` to try to merge upstream and downstream updates, we're ignoring updates from the upstream.